src/entities/Updates/routes.ts

```diff
@@ -0,0 +1,119 @@
+import { Request } from 'express'
+import routes from 'decentraland-gatsby/dist/entities/Route/routes'
+import handleAPI from 'decentraland-gatsby/dist/entities/Route/handle'
+import { auth, WithAuth } from 'decentraland-gatsby/dist/entities/Auth/middleware'
+import RequestError from 'decentraland-gatsby/dist/entities/Route/error'
+import Time from 'decentraland-gatsby/dist/utils/date/Time'
+import { getNextUpdate, getPublicUpdates, getPendingUpdates, getCurrentUpdate, isBetweenLateThresholdDate } from './utils'
+import UpdateModel from './model'
+import { UpdateAttributes, UpdateStatus } from './types'
+import ProposalModel from '../Proposal/model'
+import { ProposalAttributes } from '../Proposal/types'
+
+export default routes((route) => {
+  const withAuth = auth()
+  route.get('/proposals/:proposal/updates', handleAPI(getProposalUpdates))
+  route.get('/proposals/:update/update', handleAPI(getProposalUpdate))
+  route.post('/proposals/:proposal/update', withAuth, handleAPI(createProposalUpdate))
+  route.patch('/proposals/:proposal/update', withAuth, handleAPI(updateProposalUpdate))
+})
+
+async function getProposalUpdate(req: Request<{ update: string }>) {
+  const id = req.params.update
+
+  if (!id) {
+    throw new RequestError(`Missing id`, RequestError.NotFound)
+  }
+
+  const update = await UpdateModel.findOne<UpdateAttributes>({ id })
+
+  if (!update) {
+    throw new RequestError(`Update not found: "${id}"`, RequestError.NotFound)
+  }
+
+  return update
+}
+
+async function getProposalUpdates(req: Request<{ proposal: string }>) {
+  const proposal_id = req.params.proposal
+
+  if (!proposal_id) {
+    throw new RequestError(`Proposal not found: "${proposal_id}"`, RequestError.NotFound)
+  }
+
+  const updates = await UpdateModel.find<UpdateAttributes>({ proposal_id })
+  const publicUpdates = getPublicUpdates(updates)
```

Comment:
if the search doesn't find any updates and it returns empty/null, all following methods return an empty array?

Reply:
Yes! If there are no updates, updates will be an empty array